src/main/kotlin/org/rust/lang/core/completion/RsDeriveCompletionProvider.kt

```diff
@@ -5,25 +5,25 @@ import com.intellij.codeInsight.completion.CompletionProvider
 import com.intellij.codeInsight.completion.CompletionResultSet
 import com.intellij.codeInsight.lookup.LookupElementBuilder
 import com.intellij.patterns.ElementPattern
-import com.intellij.patterns.ElementPatternCondition
 import com.intellij.patterns.PatternCondition
 import com.intellij.patterns.PlatformPatterns.psiElement
 import com.intellij.psi.PsiElement
 import com.intellij.util.ProcessingContext
 import org.rust.lang.RsLanguage
-import org.rust.lang.core.psi.RsOuterAttr
 import org.rust.lang.core.psi.RsElementTypes.*
+import org.rust.lang.core.psi.RsOuterAttr
 import org.rust.lang.core.psi.util.parentOfType
 
-object DeriveCompletionProvider : CompletionProvider<CompletionParameters>() {
+object RsDeriveCompletionProvider : CompletionProvider<CompletionParameters>() {
 
     private val DERIVABLE_TRAITS = listOf(
         "Eq", "PartialEq",
         "Ord", "PartialOrd",
         "Copy", "Clone",
         "Hash",
         "Default",
-        "Debug"
+        "Debug",
+        "RustcEncodable", "RustcDecodable"
```

Comment:
> I have a slight inclination to the latter. I think we shouldn't prevent completion just because the user hasn't added the dependency yet.

Yes, this is the right approach I think. We need to explore auto importing stuff as well.